drivers/spi/spi_gd32.c

```diff
@@ -37,20 +41,74 @@ LOG_MODULE_REGISTER(spi_gd32);
 #else
 #error Unknown GD32 soc series
 #endif
+
+#ifdef CONFIG_SPI_GD32_DMA
+
+enum spi_gd32_dma_direction {
+	RX = 0,
+	TX,
+	NUM_OF_DIRECTION
+};
+
+struct spi_gd32_dma_config {
+	const struct device *dev;
+	uint32_t channel;
+	uint32_t config;
+	uint32_t slot;
+	uint32_t fifo_threshold;
+};
+
+struct spi_gd32_dma_data {
+	struct dma_config config;
+	struct dma_block_config block;
+	uint32_t count;
+};
+
+#endif
+
 struct spi_gd32_config {
 	uint32_t reg;
 	uint16_t clkid;
 	struct reset_dt_spec reset;
 	const struct pinctrl_dev_config *pcfg;
+#ifdef CONFIG_SPI_GD32_DMA
+	const struct spi_gd32_dma_config dma[NUM_OF_DIRECTION];
+#endif
 #ifdef CONFIG_SPI_GD32_INTERRUPT
 	void (*irq_configure)();
 #endif
 };
 
 struct spi_gd32_data {
 	struct spi_context ctx;
+#ifdef CONFIG_SPI_GD32_DMA
+	struct spi_gd32_dma_data dma[NUM_OF_DIRECTION];
+#endif
 };
 
+#ifdef CONFIG_SPI_GD32_DMA
+
+static uint32_t dummy_tx;
+static uint32_t dummy_rx;
```

Comment:
We should not access these from MCU, because DMA may access it at any time after spi driver initialized. 

SRAM can only be accessed by one AHB master at a time. So `dummy_tx` and `dummy_rx` can merge as one.

Reply:
@nandojve 
Static variable initialize default with zero. So I think it is no problem.
https://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf §6.7.8/10
(It is rare to initialize a static variable explicitly with zero in zephyr.)

@cameled 
I want to fix the rx_dummy to zero.
Transmitting modifies the value if we merge tx_dummy and rx_dummy into a single variable.
I think it may make it harder to debug.

Reply:
Hi @soburi , I update the above comment, as @nandojve suggested, we can give an explicit default zero to `dummy_tx`.

And you are right, `dummy_tx` and `dummy_rx` can not merge as one.

Reply:
Fine for me, I just rose to make sure you folks prefer 0x00 or 0xFF.